garak/probes/atkgen.py

```diff
@@ -142,6 +142,11 @@ def probe(self, generator) -> List[garak.attempt.Attempt]:
                         response = ""
                     else:
                         raise AttributeError from ae
+                except IndexError as ie:
+                    if response is None:
+                        response = ""
+                    else:
+                        raise IndexError from ie
```

Comment:
Guarding the exception may not be the best way to manage this.

If the `AttributeError` above occurs and the second attempt in line 141 returns an empty list the exception will not be captured.  Looking at how the `AttributeError` is currently handled I suspect it was due to a result from a generator that returned a list like `[ None ]` and simplification of the code to remove the try completely may be viable.

Consider:
``` Python
                # send the challenge and get the response
                response = generator.generate(challenge)
                if response is None or len(response) == 0:
                    return ""
                else:
                    return response[0].strip() if response[0] is not None else ""
```